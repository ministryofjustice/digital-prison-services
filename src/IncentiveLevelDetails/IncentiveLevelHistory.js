import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import Table from '@govuk-react/table'
import { H2 } from '@govuk-react/heading'
import { LargeScreenOnly, SmallScreenOnly } from '../Components/sizing'

const IepHistory = ({ results }) => (
  <>
    <H2>Incentive level history</H2>
    <LargeScreenOnly>
      <Table className="results">
        <Table.Row>
          <Table.CellHeader setWidth="20%">Date and time</Table.CellHeader>
          <Table.CellHeader setWidth="15%">Incentive level</Table.CellHeader>
          <Table.CellHeader setWidth="30%">Reason</Table.CellHeader>
          <Table.CellHeader setWidth="20%">Establishment</Table.CellHeader>
          <Table.CellHeader setWidth="15%">Staff member</Table.CellHeader>
        </Table.Row>
        {results.map(row => (
          <Table.Row key={row.iepTime}>
            <Table.Cell>{row.formattedTime}</Table.Cell>
            <Table.Cell>{row.iepLevel}</Table.Cell>
            <Table.Cell>{row.comments || '\u2014'}</Table.Cell>
            <Table.Cell>{row.iepEstablishment}</Table.Cell>
            <Table.Cell>{row.iepStaffMember || '\u2014'}</Table.Cell>
          </Table.Row>
        ))}
      </Table>
    </LargeScreenOnly>
    <SmallScreenOnly>
      <Table>
        <Table.Row>
          <Table.CellHeader>Date and time</Table.CellHeader>
          <Table.CellHeader>Incentive Level</Table.CellHeader>
          <Table.CellHeader>Establishment</Table.CellHeader>
        </Table.Row>
        {results.map(row => (
          <Table.Row key={row.iepTime}>
            <Table.Cell>{row.formattedTime}</Table.Cell>
            <Table.Cell>{row.iepLevel}</Table.Cell>
            <Table.Cell>{row.iepEstablishment}</Table.Cell>
          </Table.Row>
        ))}
      </Table>
    </SmallScreenOnly>
  </>
)

IepHistory.propTypes = {
  results: PropTypes.arrayOf(
    PropTypes.shape({
      comments: PropTypes.string,
      iepEstablishment: PropTypes.string.isRequired,
      iepStaffMember: PropTypes.string,
      iepTime: PropTypes.string.isRequired,
      formattedTime: PropTypes.string.isRequired,
    })
  ).isRequired,
}

const mapStateToProps = state => ({
  results: state.iepHistory.results,
})

export default connect(mapStateToProps)(IepHistory)
