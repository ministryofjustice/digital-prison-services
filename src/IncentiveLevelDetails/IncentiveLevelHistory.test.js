import React from 'react'
import testRenderer from 'react-test-renderer'
import IncentiveLevelHistory from './IncentiveLevelHistory'

const initialState = {
  iepHistory: {
    currentIepLevel: 'Standard',
    daysOnIepLevel: '625',
    currentIepDateTime: '2017-08-15T16:04:35',
    nextReviewDate: '15/08/2018',
    establishments: [{ agencyId: 'LEI', description: 'Leeds' }],
    levels: ['Standard'],
    results: [
      {
        bookingId: -1,
        iepDate: '2017-08-13',
        iepTime: '2017-08-13T16:04:35',
        formattedTime: '13/08/2017 - 16:04',
        iepEstablishment: 'Leeds',
        iepStaffMember: 'Staff Member',
        agencyId: 'LEI',
        iepLevel: 'Standard',
        userId: 'ITAG_USER',
      },
    ],
  },
}

describe('Incentive level history', () => {
  const store = {}
  const history = {}
  beforeEach(() => {
    store.getState = jest.fn()
    store.subscribe = jest.fn()
    store.dispatch = jest.fn()
    history.push = jest.fn()
    history.replace = jest.fn()
    store.getState.mockReturnValue(initialState)
  })

  it('should render the Incentive level history table correctly', () => {
    const wrapper = testRenderer.create(<IncentiveLevelHistory store={store} />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
