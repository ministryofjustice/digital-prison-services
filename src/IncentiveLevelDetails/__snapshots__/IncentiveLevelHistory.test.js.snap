// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Incentive level history should render the Incentive level history table correctly 1`] = `
Array [
  <h2
    className="sc-bZQynM fOThUN"
  >
    Incentive level history
  </h2>,
  <div
    className="sc-htoDjs xQRkw"
  >
    <table
      className="sc-EHOje ecdksn results"
    >
      <tbody
        className="sc-bwzfXH uTxCW"
      >
        <tr
          className="sc-ifAKCX cFlEyZ"
        >
          <th
            className="sc-bxivhb dnVQUz"
          >
            Date and time
          </th>
          <th
            className="sc-bxivhb hgtSCp"
          >
            Incentive level
          </th>
          <th
            className="sc-bxivhb gCiqwu"
          >
            Reason
          </th>
          <th
            className="sc-bxivhb dnVQUz"
          >
            Establishment
          </th>
          <th
            className="sc-bxivhb hgtSCp"
          >
            Staff member
          </th>
        </tr>
        <tr
          className="sc-ifAKCX cFlEyZ"
        >
          <td
            className="sc-bxivhb ckLzJk"
          >
            13/08/2017 - 16:04
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            Standard
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            â€”
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            Leeds
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            Staff Member
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <div
    className="sc-gzVnrw VtCIX"
  >
    <table
      className="sc-EHOje ecdksn"
    >
      <tbody
        className="sc-bwzfXH uTxCW"
      >
        <tr
          className="sc-ifAKCX cFlEyZ"
        >
          <th
            className="sc-bxivhb eWDuGm"
          >
            Date and time
          </th>
          <th
            className="sc-bxivhb eWDuGm"
          >
            Incentive Level
          </th>
          <th
            className="sc-bxivhb eWDuGm"
          >
            Establishment
          </th>
        </tr>
        <tr
          className="sc-ifAKCX cFlEyZ"
        >
          <td
            className="sc-bxivhb ckLzJk"
          >
            13/08/2017 - 16:04
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            Standard
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            Leeds
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
]
`;
